src/core/io_queue.cc

```diff
@@ -834,7 +834,13 @@ double internal::request_tokens(io_direction_and_length dnl, const io_queue::con
 
     const auto& m = mult[dnl.rw_idx()];
 
-    return double(m.weight) / cfg.req_count_rate + double(m.size) * (dnl.length() >> io_queue::block_size_shift) / cfg.blocks_count_rate;
+    double iops_cost = double(m.weight) / cfg.req_count_rate;
+    double tp_cost = double(m.size) * (dnl.length() >> io_queue::block_size_shift) / cfg.blocks_count_rate;
+    if (cfg.max_cost_function) {
```

Comment:
Overall comment: we should doc this somewhere and have at least one comment in the code. If none of the io-properties stuff is documented anywhere, maybe just an in-code comment somewhere good? Could point to this PR cover letter which is really good.